JS.md

```diff
@@ -173,83 +173,90 @@ is thrown. If the list of
 * If `i` is a function import:
   * If [`IsCallable(v)`](https://tc39.github.io/ecma262/#sec-iscallable) is `false`,
     throw a [`TypeError`](https://tc39.github.io/ecma262/#sec-native-error-types-used-in-this-standard-typeerror).
   * If `v` is an [Exported Function Exotic Object](#exported-function-exotic-objects):
-    * If the signature of `v` does not match the signature of `i`, a 
-      [`TypeError`](https://tc39.github.io/ecma262/#sec-native-error-types-used-in-this-standard-typeerror)
-      is thrown.
-    * Otherwise, append `v` to `imports`.
+    * If the signature of `v` does not match the signature of `i`, throw a 
+      [`TypeError`](https://tc39.github.io/ecma262/#sec-native-error-types-used-in-this-standard-typeerror).
+    * Let `closure` be `v.[[Closure]]`.
   * Otherwise:
-    * If `i` is referenced by any element of any [Elements section](Modules.md#elements-section),
-      a [`TypeError`](https://tc39.github.io/ecma262/#sec-native-error-types-used-in-this-standard-typeerror)
-      is thrown.
-    * Otherwise, append an anonymous function to `imports` 
+    * Let `closure` be a new [host function](https://github.com/WebAssembly/spec/blob/master/ml-proto/spec/instance.ml#L9)
+      of the given signature
       which calls `v` by coercing WebAssembly arguments to JavaScript arguments
-      via [`ToJSValue`](#tojsvalue) and returns the result by coercing
+      via [`ToJSValue`](#tojsvalue) and returns the result, if any, by coercing
       via [`ToWebAssemblyValue`](#towebassemblyvalue).
+  * Append `v` to `funcs`.
+  * Append `closure` to `imports`.
 * If `i` is a global import:
   * If `i` is not an immutable global, throw a [`TypeError`](https://tc39.github.io/ecma262/#sec-native-error-types-used-in-this-standard-typeerror).
   * Append [`ToWebAssemblyValue`](#towebassemblyvalue)`(v)` to `imports`.
 * If `i` is a memory import:
   * If `v` is not a [`WebAssembly.Memory` object](#webassemblymemory-objects),
     throw a [`TypeError`](https://tc39.github.io/ecma262/#sec-native-error-types-used-in-this-standard-typeerror).
-  * Otherwise, append `v.[[Memory]]` to `imports`.
+  * Append `v` to `memories`.
+  * Append `v.[[Memory]]` to `imports`.
 * Otherwise (`i` is a table import):
   * If `v` is not a [`WebAssembly.Table` object](#webassemblytable-objects),
     throw a [`TypeError`](https://tc39.github.io/ecma262/#sec-native-error-types-used-in-this-standard-typeerror).
-  * Otherwise, append `v.[[Table]]` to `imports`.
+  * Append `v` to `tables`.
+  * Append `v.[[Table]]` to `imports`.
 
 Let `instance` be the result of creating a new
-[`Eval.instance`](https://github.com/WebAssembly/spec/blob/master/ml-proto/spec/eval.ml#L15)
-given `module` and `imports`. Note: this step does not mutate imported memories
-or tables.
+[`Eval.init`](https://github.com/WebAssembly/spec/blob/master/ml-proto/spec/eval.ml#L416)
+given `module` and `imports`.
```

Comment:
How about "creating a new [Eval.instance] by calling [Eval.init] given `module` and `imports`"?